ui/src/BulkUploads.js

```diff
@@ -0,0 +1,336 @@
+          "VIEW_BULK_REFUSAL_PROGRESS"
+        )}
+        <JobDetails
+          jobTitle={detailsDialogTitle}
+          job={selectedJob}
+          showDetails={this.state.showDetails}
+          handleClosedDetails={this.handleClosedDetails}
+          onClickAway={this.handleClosedDetails}
+          onProcessJob={this.onProcessJob}
+          onCancelJob={this.onCancelJob}
+          authorisedActivities={this.state.authorisedActivities}
+          loadPermission={loadPermission}
+        ></JobDetails>
+      </div>
+    );
+  }
+
+  buildBulkProcessTable(
+    bulkJobs,
+    jobType,
+    jobTitle,
+    loadPermission,
+    viewerPermission
+  ) {
+    const bulkJobTableRows = bulkJobs.map((job, index) => (
+      <TableRow key={job.createdAt}>
+        <TableCell component="th" scope="row">
+          {job.fileName}
+        </TableCell>
+        <TableCell>{job.createdAt}</TableCell>
+        <TableCell align="right">
+          <Button
+            onClick={() => this.handleOpenDetails(job, jobType)}
+            variant="contained"
+          >
+            {convertStatusText(job.jobStatus)}{" "}
+            {[
+              "STAGING_IN_PROGRESS",
+              "VALIDATION_IN_PROGRESS",
+              "PROCESSING_IN_PROGRESS",
+            ].includes(job.jobStatus) && (
+              <CircularProgress size={15} style={{ marginLeft: 10 }} />
+            )}
+          </Button>
+        </TableCell>
+      </TableRow>
+    ));
+
+    return (
+      <div style={{ padding: 20 }}>
+        {this.state.authorisedActivities.includes(viewerPermission) && (
+          <>
+            <Typography variant="h6" color="inherit" style={{ marginTop: 20 }}>
+              {jobTitle}
+            </Typography>
+            <TableContainer component={Paper}>
+              <Table>
+                <TableHead>
+                  <TableRow>
+                    <TableCell>File Name</TableCell>
+                    <TableCell>Date Uploaded</TableCell>
+                    <TableCell align="right">Status</TableCell>
+                  </TableRow>
+                </TableHead>
+                <TableBody>{bulkJobTableRows}</TableBody>
+              </Table>
+            </TableContainer>
+          </>
+        )}
+        {this.state.authorisedActivities.includes(loadPermission) && (
+          <>
+            <input
+              accept=".csv"
+              style={{ display: "none" }}
+              id="contained-button-file"
+              type="file"
+              onChange={(e) => {
+                this.handleBulkFileUpload(e, jobType);
+              }}
+            />
+            <label htmlFor="contained-button-file">
+              <Button
+                variant="contained"
+                component="span"
+                style={{ marginTop: 10 }}
+              >
+                Upload {jobTitle} File
+              </Button>
+            </label>
+          </>
+        )}
+        <Dialog open={this.state.uploadInProgress}>
```

Comment:
there will end up being multiple dialogs... not desirable... move this into the outer render function pls